fcrepo-clients/src/main/java/edu/unc/lib/dl/event/FilePremisLogger.java

```diff
@@ -153,26 +154,49 @@ public Model getModel() {
 	@Override
 	public List<PremisEventObject> getEvents() {
 		List<PremisEventObject> events = new ArrayList<>();
-
+		ResIterator eventIt = getModel().listResourcesWithProperty(Premis.hasEventType);
 		// Find all of the events and construct a list of PremisEventObjects from them.
-		for (ResIterator eventIt = getModel().listResourcesWithProperty(Premis.hasEventType); eventIt.hasNext(); ) {
-			Resource eventResc = eventIt.nextResource();
-			PID eventPid = PIDs.get(eventResc.getURI());
-
-			// Construct a model for just this event
-			Model eventModel = ModelFactory.createDefaultModel();
-			eventModel.add(eventResc.listProperties());
-
-			// Construct the event object with a presupplied model
-			PremisEventObject event = new PremisEventObject(eventPid, repository,
-					repository.getRepositoryObjectDataLoader());
-			event.storeModel(eventModel);
-
-			events.add(event);
+		try {
+			gatherAllObjectsForEvents(eventIt, events);
+		} finally {
+			eventIt.close();
 		}
-
+		
 		log.debug("Retrieved {} events from file log for object {}",
 				events.size(), objectPid.getQualifiedId());
 		return events;
 	}
+	
+	private void gatherAllObjectsForEvents(ResIterator eventIt, List<PremisEventObject> events) {
+			while (eventIt.hasNext()) {
+				Resource eventResc = eventIt.nextResource();
+				PID eventPid = PIDs.get(eventResc.getURI());
+				// Construct a model for just this event
+				Model eventModel = ModelFactory.createDefaultModel();
+				StmtIterator stmtIt = eventResc.listProperties();
+				// Add all statements with this resc as subject to the model
+				eventModel.add(stmtIt);
+				stmtIt.close();
+				// Get a fresh iterator to check all objects of all the triples for properties
+				stmtIt = eventResc.listProperties();
+				while (stmtIt.hasNext()) {
+					RDFNode objNode = stmtIt.next().getObject();
+					if (objNode.isResource()) {
+						StmtIterator objIt = objNode.asResource().listProperties();
+						// Add statements to the event's model for any objects that have properties
+						if (objIt != null) {
+							eventModel.add(objIt);
+						}
+						objIt.close();
```

Comment:
Minor thing, but calling objIt.close here would cause an NPE if objIt were null.  Only pointing it out since there is an if statement right above it that tests for null, i'm not sure if the iterator can actually come back null.